### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by KaskMIL**
# In this new Pull Request I'm adding the following features:
- Create a classroom file that contains the Classroom class.
- Create associations has-many and belongs-to between Classroom and Student.
- Create the book and rentals file that contains Book and Rentals class.
- Create association many-to-many and has-many-through Person, Rental and book.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (rustam-tolipov) [inline book.rb:9]: - [ ] Well in that place you have to add the `add_rental(book, person)` method to your book.rb. For more information [link...](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(date, book, person)
+    @date = date
+
+    @book = book
+    book.rentals << self
+
+    @person = person
+    person.rentals << self
+  end
```
Comment (Rustamxon7) [inline rental.rb:13]: - [ ] Kindly implement the suggestion I mentioned above book.rb `add_rental(book, date)`.

File: test.rb [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
 require './trimmer_decorator'
+require './classroom'
+require './student'
+require './book'
+require './rental'
 
-person = Person.new(22, 'maximilianus')
-capitalize_person = CapitalizeDecorator.new(person)
-trim_person = TrimmerDecorator.new(capitalize_person)
+math = Classroom.new('Algebra')
```
Comment (rustam-tolipov) [inline test.rb:9]: I like the way you approach testing the project! 
However, kindly implement the suggestions I gave.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @KaskMIL, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -2,14 +2,15 @@
...
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @age = age
     @name = name
     @parent_permission = parent_permission
     @id = Random.rand(1..1000)
+    @rentals = []
```
Comment (rustam-tolipov) [inline person.rb:13]: - [ ] Kindly implement the suggestion I mentioned above book.rb `add_rental(book, date)`.

**Comment by KaskMIL** [comment]
Hi @Rustamxon7 
Thanks for your review, I will apply the changes suggested.
have a great day, cheers!

**Review by vechicin — APPROVED** [review]
# Hi @KaskMIL ✋, 

Your project is complete! 🎯 🧠 💯 
There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights ⚡ 

- [x] Good job implementing the required changes!

## Optional suggestions 📂 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**